global_helpers/panther_greynoise_helpers.py

```diff
@@ -0,0 +1,104 @@
+import ast
+import datetime
+from typing import Union
+
+from dateutil import parser
+from panther_base_helpers import deep_get
+
+
+class GreyNoise:
+    def __init__(self, event):
+        self.noise = deep_get(event, "p_enrichment", "greynoise")
+
+    # Fields available for all users
+    @property
+    def ip_address(self) -> str:
+        return self.noise.get("ip")
+
+    @property
+    def classification(self) -> str:
+        return self.noise.get("classification")
+
+    @property
+    def actor(self) -> str:
+        return self.noise.get("actor")
+
+    @property
+    def url(self) -> str:
+        return f"www.greynoise.io/viz/ip/{self.noise.get('ip')}"
+
+    # Advanced features
+    # GreyNoise Advanced Subscription
+    def _check_advanced(self, field) -> bool:
+        if self.noise.get(field) is None:
+            return False
+        return True
+
+    @property
+    def is_bot(self) -> Union[bool, str]:
+        if self._check_advanced("bot") is False:
+            return "GreyNoise Advanced Subscription required for this field"
+        return ast.literal_eval(self.noise.get("bot"))
+
+    @property
+    def cve_string(self) -> str:
+        if self._check_advanced("cve") is False:
+            return "GreyNoise Advanced Subscription required for this field"
+        cve_raw = self.noise.get("cve")
+        if isinstance(cve_raw, list):
+            return " ".join(cve_raw)
```

Comment:
Is this a typical pattern, to return a space-separated string from a list, instead of comma-separated?

Reply:
So, in some cases, these lists can be rather long.  Wondering if we want an option "limit" parameter to say "only return the first X number of entries in this list?  Same with the tag list.

Reply:
Right on, I hadn't considered those could become long lists. I'll see what I can do to introduce a limit to how much of the list is converted. Thanks for the heads up on that.